insights/parsers/dse_ldif.py

```diff
@@ -0,0 +1,240 @@
+
+    # Note about "+" options. In some rhds versions and for some keywords,
+    # a "+" at the beginning of the value list signifies that the values are
+    # added to the default list of values.
+    # IT IS NOT DETECTED AND PARSED HERE.
+    # `parse_content()` doesn't split the individual values because some
+    # keywords don't have a list of values (just one string).
+    # See the docstring in `line_uses_plus()` for more details.
+    # Re: BZ#1697477
+    # Config lines may also be delimited by `=`, and values may be quoted
+    # with `"`. Here it is assumed that config lines are well-formed.
+    def parse_content(self, content):
+        self.lines = []
+        for line in get_active_lines(content):
+            line_splits = [s.strip() for s in re.split(r"[\s:]+", line, 1)]
+            kw, val = line_splits[0], line_splits[1].strip('"') if \
+                len(line_splits) == 2 else ''
+            self.lines.append(self.KeyValue(
+                kw, val, kw.lower(), line
+            ))
+        self.keywords = set([k.kw_lower for k in self.lines])
+
+    def __contains__(self, keyword):
+        """
+        (bool) Is this keyword found in the configuration file?
+        """
+        return keyword.lower() in self.keywords
+
+    def __iter__(self):
+        """
+        (iter) Yields a list of all the lines in the configuration file that
+        actually contained a configuration option.
+        """
+        for line in self.lines:
+            yield line
+
+    def __getitem__(self, keyword):
+        """
+        (list) Get all values of this keyword.
+        """
+        kw = keyword.lower()
+        if kw in self.keywords:
+            return [kv.value for kv in self.lines if kv.kw_lower == kw]
+
+    def get(self, keyword):
+        """
+        Get all declarations of this keyword in the configuration file.
+
+        Returns:
+            (list): a list of named tuples with the following properties:
+                - ``keyword`` - the keyword as given on that line
+                - ``value`` - the value of the keyword
+                - ``kw_lower`` - the keyword converted to lower case
+                - ``line`` - the complete line as found in the config file
+        """
+        kw = keyword.lower()
+        return [kv for kv in self.lines if kv.kw_lower == kw]
+
+    def get_values(self, keyword, default='', join_with=None, split_on=None):
+        """
+        Get all the values assigned to this keyword.
+
+        Firstly, if the keyword is not found in the configuration file, the
+        value of the ``default`` option is used (defaulting to ``''``).
+
+        Then, if the ``join_with`` option is given, this string is used to
+        join the values found on each separate definition line.  Otherwise,
+        each separate definition line is returned as a string.
+
+        Finally, if the ``split_on`` option is given, this string is used to
```

Comment:
IMO, `split_on` option should be removed. From the docs, it is not clear what this is for; it complicates the method; it replicates Python's built-in functionality (a simple `.split()`).